x-pack/plugins/viz_editor/public/pseudo_plugins/xy_chart_plugin/xy_chart_plugin.tsx

```diff
@@ -153,7 +153,7 @@ function getSuggestionsForField(indexPatternName: string, field: Field, visModel
 
   if (operationNames[0] === 'terms') {
```

Comment:
This logic feels a bit suspect. The default:
```
{
        operation: operationName,
        argument: { field: field.name },
      };
```

Is not guaranteed to be valid. I wonder if we should have a TODO in here to make sure we follow up and come up with a solid way of generating the defaults here. It probably needs to be driven by the plugin.

One other (pedantic) nit pick: I find `let` statements introduce mental overhead, so I prefer to avoid them where possible. The reason is that I now have to keep track of the `firstOperation` variable all the way through the function, because who knows where it might change? Instead, this could be written something like this:

```js
function defaultOperation(field) {
  const [operationName] = getOperationsForField(field);

  if (!operationName) {
    return;
  }

  switch (operationName) {
    case 'terms':
      return {
        operation: 'terms',
        argument: { field: field.name, size: 5 },
      };
    case 'date_histogram':
      return {
        operation: 'date_histogram',
        argument: { field: field.name, interval: '1d' },
      };
    case 'count':
      return { operation: 'count' };
    default:
      return {
        operation: operationName,
        argument: { field: field.name },
      };
  }
}


function getSuggestionsForField(/* etc */) {
  const firstOperation = defaultOperation(field);

  if (!firstOperation) {
    return [];
  }

  // etc
}
```

Reply:
AFAIK Wylie is currently looking into this, including making the whole thing more stable.

Reply:
Yes, working on it